src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs

```diff
@@ -326,15 +326,18 @@ protected string CheckSourceValidity(string source)
         /// <param name="matchingSource">Matching Source if the specified source is a known source. Else null.</param>
         protected void UpdateActiveSourceRepository(string source, bool validateSource)
         {
+            // If source string is not specified, get the current active package source from the host
+            source = string.IsNullOrEmpty(source) ? (string)GetPropertyValueFromHost(ActivePackageSourceKey) : source;
```

Comment:
CheckSourceValidity is called only when validateSource bool is True. When cmdlets call UpdateActiveSourceRepository then the bool is False. Not sure if we can move it inside CheckSourceValidity.